vaadinfrontend/src/main/java/life/qbic/datamanager/views/projectOverview/ProjectOverviewHandler.java

```diff
@@ -0,0 +1,71 @@
+package life.qbic.datamanager.views.projectOverview;
+
+import com.vaadin.flow.component.Text;
+import com.vaadin.flow.data.renderer.ComponentRenderer;
+import life.qbic.projectmanagement.application.finances.offer.OfferLookupService;
+import life.qbic.projectmanagement.domain.finances.offer.OfferPreview;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Objects;
+
+/**
+ * <b>Handler</b>
+ *
+ * <p>Orchestrates the layout {@link ProjectOverviewLayout} and determines how the components behave.</p>
+ *
+ * @since <version tag>
+ */
+@Component
+public class ProjectOverviewHandler implements ProjectOverviewHandlerInterface{
+
+    private ProjectOverviewLayout registeredProjectOverview;
+    private final OfferLookupService offerLookupService;
+
+    ProjectOverviewHandler(@Autowired OfferLookupService offerLookupService){
+        Objects.requireNonNull(offerLookupService);
+        this.offerLookupService = offerLookupService;
+    }
+
+    @Override
+    public void handle(ProjectOverviewLayout layout) {
+        if (registeredProjectOverview != layout) {
+            this.registeredProjectOverview = layout;
+            addClickListeners();
+        }
+    }
+
+    private void addClickListeners() {
```

Comment:
Maybe for future implementations: the private methods can still have more meaningful names, like `configureSearchField` or similar. This helps to navigate within the class, since we are usually interested in the UI elements and their configuration here :)

Reply:
Good point, missed that. I adjusted the method